### PR TITLE
input-remapper: 1.4.2 -> 1.5.0

### PR DESCRIPTION
**Pull request by LunNova**
###### Description of changes

Minor version bump.

See https://github.com/sezanzeb/input-remapper/releases/tag/1.5.0 for changes.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/inputmethods/input-remapper/default.nix [hunk 1]
```diff
@@ -34,9 +34,9 @@
   # https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293/7
   # The names are prefixed with input_remapper to avoid potential
   # collisions with package names
-, input_remapper_version ? "1.4.2"
-, input_remapper_src_rev ? "af20f87a1298153e765b840a2164ba63b9ef937a"
-, input_remapper_src_hash ? "sha256-eG4Fx1z74Bq1HrfmzOuULQLziGdWnHLax8y2dymjWsI="
+, input_remapper_version ? "1.5.0"
+, input_remapper_src_rev ? "e31a1b2bc5d23fe13130afcc242063196335399f"
+, input_remapper_src_hash ? "sha256-KPQLgXSonuOgphagYN2JN+CMIpmjTIPUTCqOPDk0UYU="
```
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/input-remapper/default.nix:39]: Please inline this. overrideAttrs can always be used to change those.

Comment (LunNova) [reply pkgs/tools/inputmethods/input-remapper/default.nix]: Done.

File: pkgs/tools/inputmethods/input-remapper/default.nix [hunk 2]
```diff
@@ -59,9 +51,6 @@ buildPythonApplication {
   strictDeps = false;
 
   prePatch = ''
```
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/input-remapper/default.nix:53]:
```suggestion
  postPatch = ''
```

File: pkgs/tools/inputmethods/input-remapper/default.nix [hunk 3]
```diff
@@ -59,9 +51,6 @@ buildPythonApplication {
...
-    # set revision for --version output
-    echo "COMMIT_HASH = '${input_remapper_src_rev}'" > inputremapper/commit_hash.py
-
     # fix FHS paths
     substituteInPlace inputremapper/configs/data.py \
       --replace "/usr/share/input-remapper"  "$out/usr/share/input-remapper"
```
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/input-remapper/default.nix:56]: Also all the () in the line below can be removed.

File: pkgs/tools/inputmethods/input-remapper/default.nix [hunk 4]
```diff
@@ -161,4 +150,14 @@ buildPythonApplication {
...
     mainProgram = "input-remapper-gtk";
   };
-}
+}).overrideAttrs (final: prev: {
+  # Set in an override as buildPythonApplication doesn't yet support
+  # the `final:` arg yet from #119942 'overlay style overridable recursive attributes'
+  # this ensures the rev matches the input src's rev after overriding
+  # See https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293/7 for more
+  # discussion
+  prePatch = (prev.prePatch or "") + ''
```
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/input-remapper/default.nix:159]:
```suggestion
  postPatch = prev.postPatch or "" + ''
```

File: pkgs/tools/inputmethods/input-remapper/default.nix (hunk 3 above)
Comment (LunNova) [reply pkgs/tools/inputmethods/input-remapper/default.nix:56]: Done

**Comment by LunNova** [comment]
@tyriel opened a draft PR here https://github.com/NixOS/nixpkgs/pull/249777

I don't have access to the machine I normally test this on, if you're able to try it out that would be helpful.

**Pull request merged**